### PR TITLE
Poc svelte typescript

### PR DESCRIPTION
**Pull request by nickccm1122**
## What I have done:

- try using Svelte V3 alpha
- try using typescript with svelte

## Conclusion:

1. Svelte V3 alpha is a no-go at the moment

The API is different without proper documentation. For instance, `Svelte.set` is broken immediately. If we go with V3 we will have a very hard time. Besides, haven't experience better ts experience.

2. everything within a `<script>` is plain JS, no ts support. We have two ways to go if we still want to use svelte
a. use `Svelte` solely as the `View`, try to keep any complex stuff out of Svelte and in TypeScript, the coupling between the Svelte app and services could be fairly thin
b. build in JS instead

or should we just use another framework? it is my least option tho.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,6 +15,8 @@
   },
   "devDependencies": {
     "@babel/core": "^7.2.2",
+    "@babel/plugin-proposal-class-properties": "^7.3.0",
```
Comment (foray1010) [inline package.json]: same not used?

File: package.json [hunk 2]
```diff
@@ -15,6 +15,8 @@
...
+    "@babel/plugin-proposal-object-rest-spread": "^7.3.1",
```
Comment (foray1010) [inline package.json]: should be included in @babel/preset-env

Comment (nickccm1122) [reply package.json]: https://babeljs.io/docs/en/babel-preset-env#how-does-it-work

> It is important to note that @babel/preset-env does not support stage-x plugins.

**Comment by nickccm1122** [comment]
@foray1010

**Comment by nickccm1122** [comment]
@foray1010 removed those unnecessary plugins, we will add them back if needed.
also, we will have a squash merge for this PR

**Pull request merged**